src/main/java/io/cdap/delta/bigquery/BigQueryEventConsumer.java

```diff
@@ -257,20 +263,20 @@ public synchronized void applyDDL(Sequenced<DDLEvent> sequencedEvent) throws Exc
         TableId tableId = TableId.of(project, event.getDatabase(), event.getTable());
         Table table = bigQuery.getTable(tableId);
         // TODO: check schema of table if it exists already
-        // TODO: add a way to get PK of a table through Delta API?
         if (table == null) {
           TableDefinition tableDefinition = StandardTableDefinition.newBuilder()
             .setSchema(Schemas.convert(addSequenceNumber(event.getSchema())))
             .build();
-          String description;
-          if (event.getPrimaryKey().isEmpty()) {
-            description = "Primary Key: _sequence_num";
-          } else {
-            description = "Primary Key: " + event.getPrimaryKey().stream().collect(Collectors.joining(","));
+          List<String> primaryKeys = event.getPrimaryKey();
+          if (primaryKeys.isEmpty()) {
+            throw new DeltaFailureException(
+              String.format("Table '%s' in database '%s' has no primary key. Tables without a primary key are" +
+                              " not supported.", tableId.getTable(), tableId.getDataset()));
           }
-          TableInfo tableInfo = TableInfo.newBuilder(tableId, tableDefinition)
-            .setDescription(description)
-            .build();
+          primaryKeyStore.put(tableId, primaryKeys);
+          context.putState(String.format("%s-%s-primary-key", tableId.getDataset(), tableId.getTable()),
+                           GSON.toJson(primaryKeys).getBytes());
```

Comment:
what I meant here was a class that has primaryKeys as a field. That way, if we need to add more information, we can just add it to the class and things will be compatible.

Reply:
I will add a class here. One thing I want to confirm here is that due to the primary key are came together per table. Instead of writing to different file per table, I think it is better to store the map object as field into a single bigquery state file. Does it sound okay?

Reply:
One other thing we need to consider is that GCS has a rate limit of 1 update per second for the same object. So if we have a single file, it could potentially hit that rate limit as well.  With a single file, it also becomes less efficient the more tables we are tracking, as the entire state needs to be stored whenever one primary key is added/updated.

I think it's better to keep them separate.

Reply:
true. agreed. Will keep them separate.